src/charm.py

```diff
@@ -592,7 +580,13 @@ def get_secret(self, scope: Scopes, key: str) -> Optional[str]:
 
         peers = self.model.get_relation(PEER_RELATION_NAME)
```

Comment:
We should have a check here. There is an early stage in certain charm's execution when peer relation is not yet available.

```
if not peers:
    return
```

Reply:
good catch!